client/fed_test.go

```diff
@@ -851,112 +834,125 @@ func TestObjectList(t *testing.T) {
 
 	// Test that long works
 	t.Run("testPelicanObjectLsLong", func(t *testing.T) {
-		// Set path for object to upload/download
+		// Set path for ls
 		for _, export := range fed.Exports {
-			uploadURL := fmt.Sprintf("pelican://%s:%s%s", param.Server_Hostname.GetString(), strconv.Itoa(param.Server_WebPort.GetInt()), export.FederationPrefix)
-
-			// Create a pipe for output
-			r, w, _ := os.Pipe()
-			originalStdout := os.Stdout
-			os.Stdout = w
-
+			listURL, r, w, originalStdout, buf := setupObjectListTest(export)
 			go func() {
 				if export.Capabilities.PublicReads {
-					err := client.DoList(fed.Ctx, uploadURL, client.WithLongOption(true))
+					err := client.DoList(fed.Ctx, listURL, client.WithLongOption(true))
 					require.NoError(t, err)
 					w.Close()
 				} else {
-					err := client.DoList(fed.Ctx, uploadURL, client.WithLongOption(true), client.WithTokenLocation(tempToken.Name()))
+					err := client.DoList(fed.Ctx, listURL, client.WithLongOption(true), client.WithTokenLocation(tempToken.Name()))
 					require.NoError(t, err)
 					w.Close()
 				}
 			}()
-			var buf bytes.Buffer
-			_, err = io.Copy(&buf, r)
+			_, err = io.Copy(buf, r)
 			require.NoError(t, err)
 			os.Stdout = originalStdout
 
 			assert.Contains(t, buf.String(), "test")
-			// permissions for test
-			assert.Contains(t, buf.String(), "drwxrwxr-x")
+			// size for test (directories have size 0)
+			assert.Contains(t, buf.String(), "0")
+			assert.Contains(t, buf.String(), "hello_world.txt")
+			// size for hello_world.txt
+			assert.Contains(t, buf.String(), "13")
+		}
+	})
+
+	// Test that when both dironly and objectonly are specified, we work as normal
+	t.Run("testPelicanObjectLsDirOnlyAndObjectOnly", func(t *testing.T) {
+		// Set path for ls
+		for _, export := range fed.Exports {
+			listURL, r, w, originalStdout, buf := setupObjectListTest(export)
+			go func() {
+				if export.Capabilities.PublicReads {
+					err := client.DoList(fed.Ctx, listURL, client.WithObjectOnlyOption(true), client.WithDirOnlyOption(true))
```

Comment:
Does this do what it should? I'd think that passing both `withObjectOnlyOption(true)` and `withDirOnlyOption(true)` should causes an error because the two are mutually exclusive.